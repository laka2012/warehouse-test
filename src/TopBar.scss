.topbar {
    display: inline-block;
    width: 100%;
    padding: 20px 0px;
    background: burlywood;
    position: fixed;
    top: 0;
    left: 0;
    .title {
        font-size: large;
        float: left;
        margin-left: 20px;
    }
    .right-container {
        float: right;
        margin-right: 30px;
        .search-icon {
            display: inline-block;
        }
        .search {
            margin-right: 10px;
            position: relative;
            display: inline-block;
            .search-input {
                margin-left:10px;
                padding-right: 20px;
                width: 100%;
                box-sizing: border-box;
            }
            .input-clear {
                position: absolute;
                right:0; top:0;
                font-style: normal;
                font-size: 1.2em;
                cursor: pointer;
            }
            input::-ms-clear {  /* Remove IE default X */
                display: none;
            }
        }
        .cart {
            margin-top: 2px;
            text-decoration: underline;
            cursor: pointer;
            float: right;
            margin-left: 20px;
            .cart-number {
                margin-left: 10px;
            }
        }
    }
}

@media screen and (max-width: 800px) {
    .topbar {
        .right-container {
            margin-right: 0px;
        }
    }
}