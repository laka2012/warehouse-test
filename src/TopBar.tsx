import React, { Component, ChangeEvent } from 'react';
import './TopBar.scss';
import { Link } from "react-router-dom";

export interface Props {
	searchText: string;
	cartNumber: number;
	searchTextHandler: (event: ChangeEvent<HTMLInputElement>) => void;
	handleSearchClear: () => void;
}

class TopBar extends Component<Props> {
	render() {
		return (
			<div className='topbar'>
				<div className='title'>Funny Shopping Mall</div>
				<div className='right-container'>
					<div className='search-icon'><i className="fas fa-search"></i></div>
					<div className='search'>	
						<input className='search-input' placeholder='Search...' value={this.props.searchText} onChange={this.props.searchTextHandler}></input>
						{ this.props.searchText != '' &&
							<span className='input-clear' onClick={this.props.handleSearchClear}>x</span>
						}
					</div>
					<div className='cart'>
						<i className="fas fa-shopping-cart"></i><span className='cart-number'><Link to="/shoppingcart">Current Cart({this.props.cartNumber})</Link></span>
					</div>
				</div>
			</div>
		);
	}
}

export default TopBar;
