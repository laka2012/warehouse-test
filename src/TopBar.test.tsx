import React from 'react';
import { shallow } from 'enzyme';
import TopBar, {Props} from './TopBar';

function topBarTest(props: Partial<Props> = {}) {
    return (
        <TopBar {...{
            searchText:'',
            cartNumber:0,
            searchTextHandler: (e)=>{},
            handleSearchClear: ()=>{},
            ...props}}>
        </TopBar>
    );
}

describe('TopBar', () => {
	it('smoke test', () => {
		const wrapper = shallow(
			topBarTest()
		);
		expect(wrapper).toMatchSnapshot();
    });

    it('no clear button test', () => {
		const wrapper = shallow(
			topBarTest()
		);
		expect(wrapper.find('.input-clear')).toHaveLength(0);
	});
    
    it('clear button test', () => {
		const wrapper = shallow(
			topBarTest({searchText: 'test'})
		);
		expect(wrapper.find('.input-clear')).toHaveLength(1);
	});
})