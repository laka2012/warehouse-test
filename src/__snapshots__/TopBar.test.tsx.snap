// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar smoke test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopBar
    cartNumber={0}
    handleSearchClear={[Function]}
    searchText=""
    searchTextHandler={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          Funny Shopping Mall
        </div>,
        <div
          className="right-container"
        >
          <div
            className="search-icon"
          >
            <i
              className="fas fa-search"
            />
          </div>
          <div
            className="search"
          >
            <input
              className="search-input"
              onChange={[Function]}
              placeholder="Search..."
              value=""
            />
          </div>
          <div
            className="cart"
          >
            <i
              className="fas fa-shopping-cart"
            />
            <span
              className="cart-number"
            >
              <Link
                to="/shoppingcart"
              >
                Current Cart(
                0
                )
              </Link>
            </span>
          </div>
        </div>,
      ],
      "className": "topbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Funny Shopping Mall",
          "className": "title",
        },
        "ref": null,
        "rendered": "Funny Shopping Mall",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="search-icon"
            >
              <i
                className="fas fa-search"
              />
            </div>,
            <div
              className="search"
            >
              <input
                className="search-input"
                onChange={[Function]}
                placeholder="Search..."
                value=""
              />
            </div>,
            <div
              className="cart"
            >
              <i
                className="fas fa-shopping-cart"
              />
              <span
                className="cart-number"
              >
                <Link
                  to="/shoppingcart"
                >
                  Current Cart(
                  0
                  )
                </Link>
              </span>
            </div>,
          ],
          "className": "right-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-search"
              />,
              "className": "search-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-search",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="search-input"
                  onChange={[Function]}
                  placeholder="Search..."
                  value=""
                />,
                false,
              ],
              "className": "search",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search-input",
                  "onChange": [Function],
                  "placeholder": "Search...",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-shopping-cart"
                />,
                <span
                  className="cart-number"
                >
                  <Link
                    to="/shoppingcart"
                  >
                    Current Cart(
                    0
                    )
                  </Link>
                </span>,
              ],
              "className": "cart",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-shopping-cart",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/shoppingcart"
                  >
                    Current Cart(
                    0
                    )
                  </Link>,
                  "className": "cart-number",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Current Cart(",
                      0,
                      ")",
                    ],
                    "to": "/shoppingcart",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Current Cart(",
                    0,
                    ")",
                  ],
                  "type": [Function],
                },
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            Funny Shopping Mall
          </div>,
          <div
            className="right-container"
          >
            <div
              className="search-icon"
            >
              <i
                className="fas fa-search"
              />
            </div>
            <div
              className="search"
            >
              <input
                className="search-input"
                onChange={[Function]}
                placeholder="Search..."
                value=""
              />
            </div>
            <div
              className="cart"
            >
              <i
                className="fas fa-shopping-cart"
              />
              <span
                className="cart-number"
              >
                <Link
                  to="/shoppingcart"
                >
                  Current Cart(
                  0
                  )
                </Link>
              </span>
            </div>
          </div>,
        ],
        "className": "topbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Funny Shopping Mall",
            "className": "title",
          },
          "ref": null,
          "rendered": "Funny Shopping Mall",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="search-icon"
              >
                <i
                  className="fas fa-search"
                />
              </div>,
              <div
                className="search"
              >
                <input
                  className="search-input"
                  onChange={[Function]}
                  placeholder="Search..."
                  value=""
                />
              </div>,
              <div
                className="cart"
              >
                <i
                  className="fas fa-shopping-cart"
                />
                <span
                  className="cart-number"
                >
                  <Link
                    to="/shoppingcart"
                  >
                    Current Cart(
                    0
                    )
                  </Link>
                </span>
              </div>,
            ],
            "className": "right-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-search"
                />,
                "className": "search-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-search",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search-input"
                    onChange={[Function]}
                    placeholder="Search..."
                    value=""
                  />,
                  false,
                ],
                "className": "search",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search-input",
                    "onChange": [Function],
                    "placeholder": "Search...",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                false,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-shopping-cart"
                  />,
                  <span
                    className="cart-number"
                  >
                    <Link
                      to="/shoppingcart"
                    >
                      Current Cart(
                      0
                      )
                    </Link>
                  </span>,
                ],
                "className": "cart",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-shopping-cart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/shoppingcart"
                    >
                      Current Cart(
                      0
                      )
                    </Link>,
                    "className": "cart-number",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "Current Cart(",
                        0,
                        ")",
                      ],
                      "to": "/shoppingcart",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Current Cart(",
                      0,
                      ")",
                    ],
                    "type": [Function],
                  },
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
